app/controllers/tickets_controller.rb

```diff
@@ -106,4 +106,189 @@ def edit_person_validators
       dob: dob_validation_issues,
     }
   end
+
+  private def get_user_and_person
+    user_id = params[:userId]
+    wca_id = params[:wcaId]
+
+    if user_id && !wca_id
+      user = User.find(user_id)
+      person = user.person
+    elsif !user_id && wca_id
+      person = Person.find_by(wca_id: wca_id)
+      user = person.user
+    elsif user_id && wca_id
+      person = Person.find_by(wca_id: wca_id)
+      user = User.find(user_id)
+    end
+    [user, person]
+  end
+
+  private def check_errors(user, person)
+    if user.present? && person.present? && person&.user != user
+      render status: :unprocessable_entity, json: {
+        error: "Person and user not linked.",
+      }
+      return true
+    end
+
+    if user.nil? && person.nil?
+      render status: :unprocessable_entity, json: {
+        error: "User ID and WCA ID is not provided.",
+      }
+      return true
+    end
+    false
+  end
+
+  def details_before_anonymization
+    user, person = get_user_and_person
```

Comment:
Ruby methods don't start with `get_` or `set_` by convention. Plain Ruby would just be `def user_and_person`.
Since we're working in the context of Rails, where you're loading entities from a database, names like `find_user_and_person` or `user_and_person_from_params` are acceptable compromises.